### PR TITLE
Adding implementation of reversing a linked list in java

### LINKED ISSUE #1766: [Idea]: <Reversing a linked list>
### Idea Title

Reverse a linked list in java 

### Idea Description

Add implementation and documentation for reversing a linked list in Java.

### Potential Benefits

- [ ] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [ ] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by Asmi1108**
### Description
Added implementation and documentation for reversing a linked list in java!

Fixes #1766

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: java/Reverse_linked_list.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+id: Reverse_Linked_List
```
Comment (ajay-dhangar) [inline java/Reverse_linked_list.md]: replace `id: Reverse_Linked_List` with `id: reverse-linked-list`

File: java/Reverse_linked_list.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+}
+```
+
+- *Time Complexity*
+-- O(n), where n is the number of nodes.
+
+- *Space Complexity*
+--O(1), as only a constant amount of space is used.
+
```
Comment (ajay-dhangar) [inline java/Reverse_linked_list.md]: not looking in good so, can you write in good way

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: java/Reverse_linked_list.md (hunk 1 above)
Comment (Asmi1108) [reply java/Reverse_linked_list.md]: Sure!

File: java/Reverse_linked_list.md (hunk 2 above)
Comment (Asmi1108) [reply java/Reverse_linked_list.md]: Ok ok!

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by Asmi1108** [comment]
> Done

Thanks!

**Pull request merged**